reason-react-tutorial.md

```diff
@@ -0,0 +1,681 @@
+{{< /highlight >}}
+
+Now we can go convert our styles to use `bs-css`, which will generate a string that we can use as a `className`. Using `bs-css` gives a little more type safety to our css styles, if that's something that you're looking for.
+
+{{< highlight javascript >}}
+/* Inside `Card.re` */
+
+module Styles = {
+  open Css;
+  let card = style([
+    border(px(1), `solid, hex("898989")),
+    borderRadius(px(4)),
+    padding(rem(1.0))
+  ])
+};
+
+/* Later, in our `render` JSX */
+<div className={Styles.card}>
+{{< /highlight >}}
+
+Sidenote&mdash;if your IDE is yelling about an unbound module warning after you added `bs-css`, try reloading it or re-opening the file. I use VSCode and I commonly get this error after installing new Reason packages. The reason (no pun intended) has to do with the IDE loading dependencies when a file is first opened and you adding dependencies after the file was opened.
+
+That said, there's a lot of other ways to manage styles in ReasonReact! These are only two of commonly-used methods. I've personally used a [custom binding](https://github.com/benjamminj/solitaire/blob/master/src/__packages__/emotion.re) to Emotion that provides a little less type safety for style rules, but feels a little closer to the tagged template literal API.
+
+There's also a really promising project for a [PPX transform for CSS](https://github.com/astrada/ppx_bs_css). You can kinda think of a PPX transform as a Babel plugin for the Reason/OCaml language. It allows the ability to use custom syntax to describe CSS. This would allow something much closer to plain ol' CSS, without sacrificing any type-checking power. I haven't fully played with it just yet, but I've heard good things so far.
+
+For now, let's make do with `bs-css` as a styling solution, but it's always good to know that other options exist if `bs-css` isn't your cup of tea.
+
+## Managing statefulness
+
+Now, let's build the search form. We're gonna do this directly inside of `App.re` for simplicity's sake, so we'll be converting `<App />` from a stateless component to a stateful component.
+
+ReasonReact calls its' stateful components _reducer components_. In my opinion, reducer components showcase the benefit of adding the battleproof type-safety of Reason/OCaml to our React code. It's easier to sacrifice type-safety when you're writing a simple card component, but once you start adding business logic to your components that type-safety helps protect us from silly mistakes.
+
+As we dive into reducer components I find it helpful to think of the way that  [Redux](https://redux.js.org/basics/reducers) reducers work. Reducer components feel very similar to Redux, except that they're contained within the component itself instead of being connect to a global state store. If you're unfamiliar with Redux or want a refresher on how it works, check out ["Tips to learn React + Redux in 2018"](https://www.robinwieruch.de/tips-to-learn-react-redux/).
```

Comment:
A better link might be [Taming the State in React](https://roadtoreact.com) :) Need to update the other blog post for 2019!